internal/driver/driver.go

```diff
@@ -433,12 +479,17 @@ func (s *Driver) NewS7Client(deviceName string, protocol map[string]models.Proto
 }
 
 // Get S7Client by 'DeviceName'
-func (s *Driver) getS7Client(deviceName string) *S7Client {
+func (s *Driver) getS7Client(deviceName string, protocols map[string]models.ProtocolProperties) *S7Client {
 	s.mu.Lock()
 	s7Client := s.s7Clients[deviceName]
 	s.mu.Unlock()
+
 	if s7Client == nil {
-		s.lc.Errorf("Get S7Client for device [%s] error.", deviceName)
+		s.lc.Errorf("Can not get S7CLient from: %s", deviceName)
```

Comment:
I don't think this is needed. But if you still want it logged it should be a `Warnf` since you are handling it.